workflow-templates/Graphql/Rover/Graphql_Rover_K8S_Demo.yml

```diff
@@ -0,0 +1,130 @@
+name: Demo Build Workflow
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - release
+    paths-ignore:
+      - "**.md"
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    env:
+      AZURE_REGISTRY_NAME: crdsmlint5b5869c1
+      CRDS_ENV: demo
+      VAULT_URI: https://vault.crossroads.net:443
+      VAULT_ROLE_ID: ${{secrets.VAULT_ROLE_ID}}
+      VAULT_SECRET_ID: ${{secrets.VAULT_SECRET_ID}}
+      DOCKER_TAG: ${{github.sha}}
+      ASPNETCORE_ENVIRONMENT: demo
+      MP_OAUTH_BASE_URL: https://admin.crossroads.net/ministryplatformapi/oauth
+      MP_REST_API_ENDPOINT: https://admin.crossroads.net/ministryplatformapi
+      CRDS_MP_COMMON_CLIENT_ID: CRDS.Common
+      CRDS_MP_COMMON_CLIENT_SECRET: ${{secrets.CRDS_MP_COMMON_CLIENT_SECRET}}
+      DOCKER_IMAGE_SECRET: some-value-that-nobody-should-know-487374
+      NEW_RELIC_APP_NAME: ${{github.event.repository.name}}
+      NEW_RELIC_DISTRIBUTED_TRACING_ENABLED: true
+      NEW_RELIC_LICENSE_KEY: ${{secrets.NEW_RELIC_LICENSE_KEY}}
+      REPO: ${{github.event.repository.name}}
+      GRAPH: ${{secrets.SUB_GRAPH_NAME} # ADD THIS AS A REPO SECRET i.e groups
+      GRAPHQL_ENGINE_KEY: ${{secrets.GRAPHQL_ENGINE_KEY}}
+      PORT: ${{secrets.SUB_GRAPH_PORT}} #This is the localhost port that the graph runs on, set as a REPO level secret
+      APOLLO_KEY: ${{secrets.APOLLO_KEY_INT}}
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+
+      - name: Log in to Docker Hub
+        uses: docker/login-action@f054a8b539a109f9f41c372932f1ae047eff08c9
+        with:
+          username: ${{secrets.DOCKER_USERNAME}}
+          password: ${{secrets.DOCKER_PASSWORD}}
+
+      - name: Build image
+        id: docker_build
+        uses: docker/build-push-action@v2
+        with:
+          context: .
+          builder: ${{ steps.buildx.outputs.name }}
+          file: ./deployment/docker/Dockerfile
+          push: true
+          tags: crdschurch/${{env.REPO}}:${{env.DOCKER_TAG}}
+          cache-from: type=registry,ref=crdschurch/${{env.REPO}}:latest
+          cache-to: type=inline
+          build-args: |
+            GIT_ACCESS_TOKEN=${{ secrets.ACCESS_TOKEN }}
+            VAULT_ROLE_ID=${{ env.VAULT_ROLE_ID }}
+            VAULT_SECRET_ID=${{ env.VAULT_SECRET_ID }}
+            CRDS_ENV=${{ env.CRDS_ENV }}
+      - name: Connect to Azure
+        uses: azure/docker-login@v1
+        with:
+          login-server: ${{env.AZURE_REGISTRY_NAME}}.azurecr.io
+          username: ${{secrets.AZURE_USER}}
+          password: ${{secrets.AZURE_PASSWORD}}
+
+      - name: Set Azure context
+        uses: azure/aks-set-context@v1
+        with:
+          creds: '{
+            "clientId": "${{secrets.AZURE_CLIENT_ID_DEMO}}",

```

Comment:
For Azure Demo and INT share an environment. For the Azure Context the values would be the same as INT:

 "clientId": "${{secrets.AZURE_CLIENT_ID}}",
              "clientSecret": "${{secrets.AZURE_CLIENT_SECRET}}",
              "subscriptionId": "${{secrets.AZURE_SUBSCRIPTION_INT_DEMO}}",
              "tenantId": "${{secrets.AZURE_TENANT}}",

Reply:
These values are different for PROD: 

"clientId": "${{secrets.AZURE_CLIENT_ID_PROD}}",
              "clientSecret": "${{secrets.AZURE_CLIENT_SECRET_PROD}}",
              "subscriptionId": "${{secrets.AZURE_SUBSCRIPTION_PROD}}",
              "tenantId": "${{secrets.AZURE_TENANT}}",